compose/ui/ui/src/jsWasmMain/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt

```diff
@@ -189,12 +184,22 @@ private class ComposeWindow(
         }
     }
 
-    fun resize(newSize: IntSize) {
-        canvas.width = newSize.width
-        canvas.height = newSize.height
-        _windowInfo.containerSize = IntSize(canvas.width, canvas.height)
+    fun resize(boxSize: IntSize) {
+        val density = density.density.toInt()
+
+        val width = boxSize.width * density
+        val height = boxSize.height * density
+
+        // TODO: What we actually should do is to pass this new dimensions to attachTo without setting them directly on canvas.
+        // We should be HTMLCanvas-agnostic in that sense
```

Comment:
I'm not sure we should aim to be HTMLCanvas-agnostic in CavnasBasedWindow implementations.
It's easier to implement the details here rather than in skiko, IMO.

Reply:
by agnostic I meant that ideally there should be only one place where we are setting canvas dimensions physically. Now we do it in three different places and this can get out of hand. If this place will be somewhere in compose this is indeed an issue of design. But I'd rather definitely width/height setting to one place.